src/test/java/com/example/demo/service/EmployeeServiceTest.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    @Test(expected = ParseException.class)
+    public void testGetAllEmployees_ThrowsParseException() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenThrow(ParseException.class);
+        // Run the test
+        employeeServiceUnderTest.getAllEmployees();
+    }
+
+    @Test
+    public void testGetAllEmployeesWhereSalaryGreater() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenReturn(MockDataProvider.getListOfMockEmployee());
+        // Run the test
+        final List<EmployeeEntity> result = employeeServiceUnderTest.getAllEmployeesWhereSalaryGreater(12000);
+        // Verify the results
+        assertEquals(result.size(), 2);
+    }
+
+    @Test(expected = IOException.class)
+    public void testGetAllEmployeesWhereSalaryGreater_ThrowsIOException() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenThrow(IOException.class);
+        // Run the test
+        employeeServiceUnderTest.getAllEmployeesWhereSalaryGreater(0);
+    }
+
+    @Test(expected = ParseException.class)
+    public void testGetAllEmployeesWhereSalaryGreater_ThrowsParseException() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenThrow(ParseException.class);
+        // Run the test
+        employeeServiceUnderTest.getAllEmployeesWhereSalaryGreater(0);
+    }
+
+    @Test
+    public void testGetEmployeeWithLowestSalary() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenReturn(MockDataProvider.getListOfMockEmployee());
+        // Run the test
+        final List<EmployeeEntity> result = employeeServiceUnderTest.getEmployeeWithSalary(false);
+
+        // Verify the results
+        assertEquals(result.get(0).getEmployeeSalary(), 12000);
+    }
+
+    @Test(expected = IOException.class)
+    public void testGetEmployeeWithSalary_ThrowsIOException() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenThrow(IOException.class);
+        // Run the test
+        employeeServiceUnderTest.getEmployeeWithSalary(false);
+    }
+
+    @Test(expected = ParseException.class)
+    public void testGetEmployeeWithSalary_ThrowsParseException() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenThrow(ParseException.class);
+        // Run the test
+        employeeServiceUnderTest.getEmployeeWithSalary(false);
+    }
+
+    @Test
+    public void testGetEmployeeWithHighestSalary() throws Exception {
+        // Setup
+        when(employeeRepository.getAllEmployees()).thenReturn(MockDataProvider.getListOfMockEmployee());
+        // Run the test
+        final List<EmployeeEntity> result = employeeServiceUnderTest.getEmployeeWithSalary(true);
+
+        // Verify the results
+        assertEquals(result.get(0).getEmployeeSalary(), 18000);
+    }
+}
```

Comment:
You can add screenshots of the test results in the Pull Request. 
BTW, Just checking , could you confirm you are pushing the code only after verifying the build is success, right ?

Reply:
Yes after a successful build, I push the code.